### ISSUE TITLE
Perf: InteriorPointArea - bisector intersection with geometry

### ISSUE THREAD

**Issue by jandam** [locationtech/jts#176]
Geometry intersections = bisector.intersection(geometry);
in InteriorPointArea::addPolygon uses general intersection algorithm. We know that bisector is horizontal LineSegment. So there can be specialized algorithm for horizontal/vertical lines. INTERSECTION in OverlayOp can be improved to use segments that fits to Envelope intersection from both geometries. I think that other segments/monotone chains can be omitted.
Example: polygon of Czech Republic
~80k nodes
-- following can be eliminated if we use only segments/monotone chains that intersects with horizontal line
~22k monotone chains
~44k Mark&Sweep events

**Comment by dr-jts** [comment]
Good point, using the general intersection algorithm in `InteriorPointArea `is quite inefficient.  

I think this would be improved with a simpler customized `HorizontalLineIntersection `algorithm, rather than trying to change the current overlay algorithm.   Of course, this could be provided as a branch in `Geomery.intersection` by checking for an input of a horizontal line.  

On the wish list is a ClipToRectangle algorithm, which might provide this capability as well.  Although I suspect that an algorithm specifically for horizontal lines could be even simpler.

**Comment by dr-jts** [comment]
@jandam Is the OSM polygon dataset publicly available?

**Comment by jandam** [comment]
Here is polygon of Czech Republic border in WKB format (WGS84) extracted from OpenStreetMap. OSM must be mentioned according to licence. 
http://www.openstreetmap.org/copyright
Source data in PBF format for whole country can be downloaded from 
http://download.geofabrik.de/europe/czech-republic.html#


[czech-republic-poly.zip](https://github.com/locationtech/jts/files/1313275/czech-republic-poly.zip)
